packages/typescript-client/src/parser.ts

```diff
@@ -0,0 +1,74 @@
+import { ColumnInfo, Value } from './types'
+
+export type ParseFunction = (
+  value: string,
+  additionalInfo?: Omit<ColumnInfo, `type` | `dims`>
+) => Value
+export type Parser = { [key: string]: ParseFunction }
+
+const parseNumber = (value: string) => Number(value)
+const parseBool = (value: string) => value === `true` || value === `t`
+const parseBigInt = (value: string) => BigInt(value)
+const parseJson = (value: string) => JSON.parse(value)
+
+export const defaultParser: Parser = {
```

Comment:
no dates? It seems like most (all?) postgres dates could be converted to a Date object.

Reply:
I think we want to stay away from Dates as turning PG dates into JS dates can be confusing.
If you have a timestamp with time zone it's easy, but if you have for instance just a postgres `Date` then what do you do? Do you turn it into a JS date and set the time to midnight? Midnight local time or midnight UTC? 

Similarly, if you have a PG `time` without a date, what do you do? You add today's date? You add the first epoch day (1st jan. 1970)?

What do you do with Postgres timestamps without time zone? Do you parse them as that time in the client's current time zone?

I think it's best we are not opinionated and let clients decide how they want to handle it.
At most, i would add timestamps with time zones as those are pretty clear how to handle.

Reply:
cool yeah makes sense. And people can easily add their own custom parsing now so if it's straightforward to get Date objects, then they can do it.